src/main/java/com/example/demo/service/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,80 @@
+package com.example.demo.service;
+
+import com.example.demo.dto.shoppingcart.CartItemRequestDto;
+import com.example.demo.dto.shoppingcart.ShoppingCartResponseDto;
+import com.example.demo.mapper.CartItemMapper;
+import com.example.demo.mapper.ShoppingCartMapper;
+import com.example.demo.model.Book;
+import com.example.demo.model.CartItem;
+import com.example.demo.model.ShoppingCart;
+import com.example.demo.model.User;
+import com.example.demo.repository.BookRepository;
+import com.example.demo.repository.CartItemRepository;
+import com.example.demo.repository.ShoppingCartRepository;
+import com.example.demo.repository.UserRepository;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private ShoppingCartRepository shoppingCartRepository;
+    private CartItemRepository cartItemRepository;
+    private ShoppingCartMapper shoppingCartMapper;
+    private CartItemMapper cartItemMapper;
+    private UserRepository userRepository;
+    private BookRepository bookRepository;
+
+    @Override
+    public ShoppingCart createShoppingCart(Long userId) {
+        User user = userRepository.findById(userId)
```

Comment:
```suggestion
```
remove as you use this method in UserServiceImpl where you register and save the user